overlord/devicestate/handlers_systems.go

```diff
@@ -318,11 +312,6 @@ func (m *DeviceManager) undoCreateRecoverySystem(t *state.Task, _ *tomb.Tomb) er
 }
 
 func (m *DeviceManager) doFinalizeTriedRecoverySystem(t *state.Task, _ *tomb.Tomb) error {
-	if release.OnClassic {
-		// TODO: this may need to be lifted in the future
-		return fmt.Errorf("internal error: cannot finalize recovery systems on a classic system")
-	}
```

Comment:
is the thinking here to have not check because anyway if the tasks are setup wrongly we will fail anyway first  doCreateRecoverySystem?